File: backend/src/main/java/com/declaration/entity/CityInfo.java
package com.declaration.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 城市信息实体类
 */
@Data
@TableName("city_info")
public class CityInfo {

    /**
     * 主键ID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /**
     * 城市中文名
     */
    private String cityName;

    /**
     * 城市英文名
     */
    private String cityEnglishName;

    /**
     * 省份中文名
     */
    private String provinceName;

    /**
     * 省份英文名
     */
    private String provinceEnglishName;

    /**
     * 国家中文名
     */
    private String countryName;

    /**
     * 国家英文名
     */
    private String countryEnglishName;

    /**
     * 城市编码
     */
    private String cityCode;

    /**
     * 排序
     */
    private Integer sort;

    /**
     * 状态（1-启用，0-禁用）
     */
    private Integer status;

    /**
     * 备注
     */
    private String remarks;

    /**
     * 创建者
     */
    private String createBy;

    /**
     * 创建时间
     */
    private LocalDateTime createTime;

    /**
     * 更新者
     */
    private String updateBy;

    /**
     * 更新时间
     */
    private LocalDateTime updateTime;

}

File: frontend/src/api/system/city-info.ts
import request from '@/utils/request'

export interface CityInfo {
  id?: number
  cityName?: string
  cityEnglishName?: string
  provinceName?: string
  provinceEnglishName?: string
  countryName?: string
  countryEnglishName?: string
  cityCode?: string
  sort?: number
  status?: number
  remarks?: string
}

export interface CityQueryParams {
  country?: string
  province?: string
}

/**
 * 获取城市信息列表
 */
export const getCities = (params?: CityQueryParams) => {
  return request({
    url: '/v1/cities',
    method: 'get',
    params
  })
}

/**
 * 获取城市信息列表
 */
export const getCitiesByCountry = (countryName?: string) => {
  return request({
    url: `/v1/cities/${countryName}/cities`,
    method: 'get',
  })
}

/**
 * 根据ID获取城市信息
 */
export const getCityById = (id: number) => {
  return request({
    url: `/v1/cities/${id}`,
    method: 'get'
  })
}

/**
 * 获取城市信息列表（分页）
 */
export const getCityList = (params?: any) => {
  return request({
    url: '/v1/cities',
    method: 'get',
    params
  })
}

/**
 * 搜索城市
 */
export const searchCities = (keyword: string) => {
  return request({
    url: `/v1/cities/search`,
    method: 'get',
    params: { keyword }
  })
}

/**
 * 新增城市信息
 */
export const addCity = (data: CityInfo) => {
  return request({
    url: '/v1/cities',
    method: 'post',
    data
  })
}

/**
 * 修改城市信息
 */
export const updateCity = (id: number, data: CityInfo) => {
  return request({
    url: `/v1/cities/${id}`,
    method: 'put',
    data
  })
}

/**
 * 删除城市信息
 */
export const deleteCity = (id: number) => {
  return request({
    url: `/v1/cities/${id}`,
    method: 'delete'
  })
}

/**
 * 更新城市状态
 */
export const updateCityStatus = (id: number, status: number) => {
  return request({
    url: `/v1/cities/${id}/status`,
    method: 'put',
    params: { status }
  })
}

/**
 * 获取城市省份列表
 */
export const getCityProvinces = () => {
  return request({
    url: '/v1/cities/provinces',
    method: 'get'
  })
}